hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -2165,13 +2167,27 @@ public List<WAL> getWALs() {
 
   @Override
   public WAL getWAL(RegionInfo regionInfo) throws IOException {
-    WAL wal = walFactory.getWAL(regionInfo);
+    WAL wal = regionBelongsToReplicatedTable(regionInfo) ?
+      walFactory.getWALInGlobalScope(regionInfo) : walFactory.getWAL(regionInfo);
```

Comment:
I guess this does the trick of separating replicable/non-replicable data in different wal files. Mind explain how the replication source reader gets only the wal with replicable data?

Reply:
Ok, so answer for my question above is [here](https://github.com/apache/hbase/pull/6532/files#diff-35e5384a3d8da9cab3138b1f1411d43bed05856e2b8951f1c4c195049503486fR54).